cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -196,6 +196,23 @@ struct ARROW_EXPORT S3Options {
   /// delay between retries.
   std::shared_ptr<S3RetryStrategy> retry_strategy;
 
+  /// the SSE-C customized key (raw 32 bytes key).
+  std::string sse_customer_key;
+
+  /// Path to a single PEM file holding all TLS CA certificates
+  ///
+  /// If empty, the underlying TLS library's defaults will be used.
+  std::string tls_ca_file_path;
+
+  /// Path to a directory holding TLS CA certificates in individual PEM files
+  /// named along the OpenSSL "hashed" format.
+  ///
+  /// If empty, the underlying TLS library's defaults will be used.
+  std::string tls_ca_dir_path;
+
+  /// Controls whether to verify SSL certificates, Default to true
+  bool tls_verify_certificates = true;
```

Comment:
Could you update `S3Options::FromUri()` to accept them?
Could you also add tests for them?
See also: https://github.com/apache/arrow/blob/d4516c5386f84619dfdf2a9f72fed6d7df89704c/cpp/src/arrow/filesystem/s3fs_test.cc#L257-L321

Or how about creating a separated PR that includes only TLS related `S3Options` and `FileSystemGlobalOptions` changes?
FYI: I created https://github.com/apache/arrow/issues/44333 yesterday.

Reply:
ok, I've add the ut in the last commit.